### PR TITLE
feat: Override default property for style helpers

### PR DESCRIPTION
**Pull request by padawannn**

### CODE REVIEW

File: src/lib/viz/style/helpers/color-bins-style.ts [hunk 1]
```diff
@@ -21,6 +22,8 @@ export interface ColorBinsOptionsStyle extends Partial<BasicOptionsStyle> {
   nullColor: string;
   // Color applied to features which the attribute value is not in the breaks.
   othersColor: string;
+  // Styling property.
+  property?: ColorsProperties;
```
Comment (VictorVelarde) [inline src/lib/viz/style/helpers/color-bins-style.ts]: Nice!

File: src/lib/viz/style/helpers/color-categories-style.ts [hunk 2]
```diff
@@ -19,6 +20,8 @@ export interface ColorCategoriesOptionsStyle extends Partial<BasicOptionsStyle>
   nullColor: string;
   // Color applied to features which the attribute value is not in the breaks.
   othersColor: string;
+  // Styling property.
+  property?: ColorsProperties;
```
Comment (VictorVelarde) [inline src/lib/viz/style/helpers/color-categories-style.ts]: Nit. I would probably go for singular naming here for the type, so `ColorProperty` (and the same with `SizeProperty` type). What do you think?

File: src/lib/viz/style/helpers/size-categories-style.ts [hunk 3]
```diff
@@ -112,12 +115,16 @@ function calculateWithCategories(
...
-    };
+    if (!options.property || options.property === 'size') {
+      obj = {
+        getRadius: getSizeValue,
+        pointRadiusMinPixels: minSize,
+        pointRadiusMaxPixels: maxSize,
+        radiusUnits: 'pixels'
+      };
+    } else if (options.property === 'strokeWidth') {
+      obj = { getLineWidth: getSizeValue };
```
Comment (VictorVelarde) [inline src/lib/viz/style/helpers/size-categories-style.ts]: Should we set the line widht Units param to pixels too?

**Review by VictorVelarde — COMMENTED** [review]
@padawannn Overall it looks nice. I just did a comment on naming. 

I think we should probably add some basic unit tests to validate 'correct' inputs and deckgl functions set on every case. And also a simple example in the examples/debug/styles folder for playing a bit with it (we could later decide if we could promote that example to the public folder)

File: src/lib/viz/style/helpers/color-categories-style.ts (hunk 2 above)
Comment (padawannn) [reply src/lib/viz/style/helpers/color-categories-style.ts]: Done

File: src/lib/viz/style/helpers/size-categories-style.ts (hunk 3 above)
Comment (padawannn) [reply src/lib/viz/style/helpers/size-categories-style.ts]: Done

**Comment by padawannn** [comment]
The example is https://github.com/CartoDB/web-sdk/pull/98/files#diff-4b7e7f58a59ba200eb20aeb64093a158 and I've added some test for each helper

File: examples/_debug/styles/color-bins-style- override-property.html [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+          strokeWidth: 4,
+          color: '#FFF',
+          property: 'strokeColor'
+        });
+        const layer = new carto.viz.Layer('listings_madrid', style);
+        layer.addTo(deckMap);
+      }
+
+      function initializeLines() {
+        carto.auth.setDefaultCredentials({ username: 'cartoframes' });
```
Comment (VictorVelarde) [inline examples/_debug/styles/color-bins-style- override-property.html:64]: Just a comment. 👍 This is OK for debug. For public examples, we should always use 'public' user

Comment (VictorVelarde) [reply examples/_debug/styles/color-bins-style- override-property.html:64]: By the way, the example looks really nice. 

It is not a priority right now, but we would probably want to document this new feature a bit better in the Styles guide, with a note like:

>>> With color helpers, the surface of the feature is coloured by default (thus, there is an implicit visual property: `color`, which we considered the most common one). But if you want to use the stroke, you can specify it, with something like for example:

```js
carto.viz.style.colorBins('availability_365', { strokeWidth: 4, property: 'strokeColor' });
```

image.

And the same could be applied for size.

cc/ @borja-munoz

Comment (VictorVelarde) [reply examples/_debug/styles/color-bins-style- override-property.html:64]: Every step we take on this direction, we're near this 'advanced VL' example: https://carto.com/developers/carto-vl/examples/maps/advanced/taxi-pickups.html which is to me one of the most complex examples I remember: several properties at the same time (+ animation)

File: src/lib/viz/__tests__/styles/ColorBinsStyle.spec.ts [hunk 5]
```diff
@@ -131,6 +131,31 @@ describe('ColorBinsStyle', () => {
         expect(error).toBeInstanceOf(CartoStylingError);
       }
     });
+
+    it('If geometryType is Polygon and property is strokeColor getLineColor should be a function', () => {
```
Comment (VictorVelarde) [inline src/lib/viz/__tests__/styles/ColorBinsStyle.spec.ts:135]: Really nice!

File: src/lib/viz/__tests__/styles/ColorCategoriesStyle.spec.ts [hunk 6]
```diff
@@ -127,6 +127,36 @@ describe('ColorCategoriesStyle', () => {
         expect(error).toBeInstanceOf(CartoStylingError);
       }
     });
+
+    it('If geometryType is Polygon and property is strokeColor getLineColor should be a function', () => {
```
Comment (VictorVelarde) [inline src/lib/viz/__tests__/styles/ColorCategoriesStyle.spec.ts:131]: 👌

**Review by VictorVelarde — APPROVED** [review]
Really nice work. Add a line in the changelog like this, and we're ready to go 🚀 

###Added
- New option to specify the visual property to use in color ('color' | 'strokeColor') and size ('size' | 'strokeWidth') style helpers ([#98](https://github.com/CartoDB/web-sdk/pull/98))

**Comment by padawannn** [comment]
@VictorVelarde changelog updated

**Pull request merged**